src/Microsoft.Diagnostics.EventFlow.Outputs.EventHub/EventHubOutput.cs

```diff
@@ -187,23 +182,53 @@ private async void CleanUpAsync()
             await Task.WhenAll(tasks).ConfigureAwait(false);
         }
 
-        private IEventHubClient CreateEventHubClient(string connectionString)
+        private IEventHubClient CreateEventHubClient(EventHubOutputConfiguration _)
         {
-            Debug.Assert(this.outputConfiguration.ConnectionString != null);
-            EventHubsConnectionStringBuilder connStringBuilder = new EventHubsConnectionStringBuilder(this.outputConfiguration.ConnectionString);
+            Debug.Assert(this.outputConfiguration != null);
+
+            if (this.outputConfiguration.UseAzureIdentity)
+            {
+                this.eventHubName = this.outputConfiguration.EventHubName;
+                ensureEventHubName();
+
+                if (string.IsNullOrWhiteSpace(this.outputConfiguration.FullyQualifiedNamespace))
+                {
+                    var emptyNamespaceMsg = $"{nameof(EventHubOutput)}: Event Hub namespace must not be empty when using Azure Identity. It can be specified in the '{nameof(EventHubOutputConfiguration.FullyQualifiedNamespace)}' configuration parameter";
+                    healthReporter.ReportProblem(emptyNamespaceMsg, EventFlowContextIdentifiers.Configuration);
+                    throw new Exception(emptyNamespaceMsg);
+                }
+
+                return new EventHubClientImpl(
+                    new EventHubProducerClient(this.outputConfiguration.FullyQualifiedNamespace, this.eventHubName, new DefaultAzureCredential())
```

Comment:
So, this is kind of sketchy, but since EventFlow configured via "json file" I am not sure we need to expose `ITokenCredential`.

Reply:
The design principle with EventFlow is that JSON configuration is for common use cases, and programmatic configuration is for advanced use cases. It is OK for certain configurations to be doable only through programmatic configuration, but the opposite should never be the case.

Consider adding `Credential` property to `EventHubOutputConfiguration` and using it here.

How common is it to use a non-default credential?

Reply:
It's generally pointless to have "DefaultAzureCredential" in Azure since in Azure you only need "ManagedAzureCredential".

But locally, "DefaultAzureCredential" is a way to go since it includes everything (your Visual Studio credentials, Azure CLI, Managed Identity etc).

I think exposing _optional_ `ITokenCredential` is a good idea.

Reply:
Updated to include `TokenCredential`. Also, based on docs, having `DefaultTokenCredential` should be fine: https://docs.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme#defaultazurecredential

Reply:
great, thanks